tests/host_tools/memory.py

```diff
@@ -3,10 +3,12 @@
 """Utilities for measuring memory utilization for a process."""
 from queue import Queue
 import time
-from threading import Thread
+from threading import Thread, Lock
 
 from framework import utils
 
+data_lock = Lock()
```

Comment:
why don't we store this variable on the `MemoryMonitor` object? AFAICT it should be per-thread instead of global

Reply:
yes, indeed I can move it to the MemMonitor thread.

Reply:
I pushed the change, PTAL, ty